charts/sentry-apps/templates/application.yaml

```diff
@@ -20,33 +20,41 @@ spec:
       parameters:
         - name: timeout
           value: 10m
-      values: |-
-        user: {{ .Values.user | toJson }}
-
-        auth:
-          register: true
-        ingress:
-          enabled: {{ .Values.ingress.hostname | empty | not }}
-          hostname: {{ .Values.ingress.hostname | default "" | quote }}
-
-        postgresqlPassword: {{ .Values.sentry.postgresqlPassword | quote }}
-        system:
-          secretKey: {{ .Values.system.secretKey | quote }}
-
-        clickhouse:
-          enabled: true
-          clickhouse: {{ .Values.clickhouse | toJson }}
-        kafka: {{ .Values.kafka | toJson }}
-        postgresql:
-          persistence:
-            enabled: true
-        rabbitmq:
-          enabled: {{ .Values.rabbitmq.enabled }}
-        redis:
-          cluster:
-            enabled: false
-        sentry:
-          worker: {{ .Values.sentry.worker | toJson }}
-        zookeeper:
-          persistence:
-            enabled: true
+        {{- range $k, $v := .Values.user }}
+        - name: user.{{ $k }}
+          value: {{ $v }}
+        {{- end }}
+        {{- if .Values.ingress.hostname | empty | not }}
+        - name: ingress.enabled
+          value: true
+        - name: ingress.hostname
+          value: {{ .Values.ingress.hostname | default "" | quote }}
+        {{- end }}
+        - name: postgresqlPassword
+          value: {{ .Values.sentry.postgresqlPassword | quote }}
+        - name: system.secretKey
+          value: {{ .Values.system.secretKey | quote }}
+        {{- range $k, $v := .Values.clickhouse }}
+        - name: clickhouse.clickhouse.{{ $k }}
+          value: {{ $v }}
+        {{- end }}
+        {{- range $k, $v := .Values.kafka }}
+        - name: kafka.{{ $k }}
+          value: {{ $v }}
+        {{- end }}
+        {{- range $k, $v := .Values.kafka }}
+        - name: kafka.{{ $k }}
+          value: {{ $v }}
+        {{- end }}
+        {{- range $k, $v := .Values.sentry.worker }}
+        - name: sentry.worker.{{ $k }}
+          value: {{ $v }}
+        {{- end }}
+        - name: postgresql.persistence.enabled
+          value: true
+        - name: rabbitmq.enabled
+          value: {{ .Values.rabbitmq.enabled }}
+        - name: redis.cluster.enabled
+          value: false
+        - name: zookeeper.persistence.enabled
+          value: true
```

Comment:
@eyenx I've had to perform this change. @tongpu is telling me that you've been experience something alike. What do you think?

Reply:
For another cluster we were kinda lazy and just did this:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: dashboard
  namespace: {{ .Values.argocd.namespace }}
spec:
  destination:
    namespace: {{ .Values.kubernetesDashboard.destination.namespace }}
    server: {{ .Values.spec.destination.server }}
  project: {{ .Values.kubernetesDashboard.project }}
  source:
    chart: {{ .Values.kubernetesDashboard.source.chart }}
    repoURL: {{ .Values.kubernetesDashboard.source.repoURL }}
    targetRevision: {{ .Values.kubernetesDashboard.source.targetRevision }}
    helm:
      values: |-
        {{- .Values.kubernetesDashboard | toYaml | nindent 8 }}
```

Reply:
At this point, being really lazy would translate to `{{ . Values | toYaml }}` :wink:

Reply:
That's exactly what we do on some other clusters.

Reply:
The `{{- .Values.kubernetesDashboard | toYaml | nindent 8 }}` is not primarily due to laziness, it also makes sure that customizations don't show in the Argo CD ui as change. The parameters struct can then be used to do a quick and dirty override through through the GUI and that will then show up as :warning: in the UI.

Reply:
From my understanding, anything in `values` doesn't apply as _override_.

Reply:
I got it working.